Source/items.cpp

```diff
@@ -521,22 +521,38 @@ void CalcPlrItemMin(Player &player)
 	}
 }
 
+uint8_t CalculateMagicRequirementForBook(spell_id spell, int8_t spellLevel)
+{
+	int magicRequirement = spelldata[spell].sMinInt;
+
+	while (spellLevel != 0) {
+		magicRequirement += 20 * magicRequirement / 100;
+		spellLevel--;
+		if (magicRequirement + 20 * magicRequirement / 100 > 255) {
+			magicRequirement = 255;
+			break;
+		}
+	}
+
+	return magicRequirement;
+}
+
 void CalcPlrBookVals(Player &player)
 {
-	for (Item &item : InventoryPlayerItemsRange { player }) {
+	auto processItem = [](Player &plr, Item &item) {

```

Comment:
This would be the only place in the code where we write `plr` instead of `player`
```suggestion
	auto processItem = [](Player &player, Item &item) {
```

Reply:
In C#, Visual Studio would have yelled at me for using the same variable name. I've been conditioned. 😓